src/coord/src/catalog.rs

```diff
@@ -2004,10 +2013,41 @@ impl Catalog {
         })
     }
 
-    /// Returns a mapping that indicates all indices that are available for
-    /// each item in the catalog.
-    pub fn indexes(&self) -> &HashMap<GlobalId, Vec<(GlobalId, Vec<MirScalarExpr>)>> {
-        &self.indexes
+    /// Returns a mapping that indicates all indices that are available for each
+    /// item in the catalog.
+    ///
+    /// Note that when `self.config.disable_user_indexes` is `true`, this does
+    /// not include any user indexes.
+    pub fn enabled_indexes(&self) -> &HashMap<GlobalId, Vec<(GlobalId, Vec<MirScalarExpr>)>> {
+        &self.enabled_indexes
+    }
+
+    /// Returns whether or not the specified index is enabled.
+    ///
+    /// The relationship between the `index` and the `id` are the caller's
+    /// responsibility to verify.
+    ///
+    /// Indexes are only disabled when Materialize is booted with
+    /// `--disable-user-indexes` and they're user indexes.
+    pub fn is_index_enabled(&self, index: &Index, id: &GlobalId) -> bool {
+        self.enabled_indexes()
+            .get(&index.on)
+            .expect("object known to exist")
+            .iter()
+            .any(|(idx_id, _)| idx_id == id)
+    }
+
+    /// Returns all of the enabled indexes for the table specified with `id`.
+    ///
+    /// # Panics
+    /// If the provided `id` does not correspond to a table.
+    pub fn get_tables_enabled_indexes(
```

Comment:
derp: this should be "object" not "table" but the shape would remain the same.

Reply:
Does this need to exist? `catalog.enabled_indexes()[id]` seems ok to me?

Reply:
This was in response to some [feedback](https://github.com/MaterializeInc/materialize/pull/7887#discussion_r690708832) from Matt; I have a preference for axing it, but wanted to show what this looked like